arduino/smartcar/smartcar.ino

```diff
@@ -1,9 +1,264 @@
+            // initial one (e.g. started at 350 degrees and now we are at 20), so to get a
+            // signed displacement (+30)
+            currentHeading -= 360;
+        }
+        // Degrees turned so far is initial heading minus current (initial heading
+        // is at least 0 and at most 360. To handle the "edge" cases we substracted or added 360
+        // to currentHeading)
+        degreesTurnedSoFar = (degrees > 0 ? initialHeading - currentHeading : currentHeading - initialHeading);
+
+        hasReachedTargetDegrees = smartcarlib::utils::getAbsolute(degreesTurnedSoFar)
+                                  >= smartcarlib::utils::getAbsolute(degrees) - 5;
+    }
+    car.setSpeed(0);
+}
+
+void go(long centimeters, float speed)
+{
+    if (centimeters == 0)
+    {
+        return;
+    }
+    // Ensure the speed is towards the correct direction
+    speed = smartcarlib::utils::getAbsolute(speed) * ((centimeters < 0) ? -1 : 1);
+    car.setSpeed(speed);
+    car.setAngle(0);
+
+    long initialDistance          = car.getDistance();
+    bool hasReachedTargetDistance = false;
+    while (!hasReachedTargetDistance)
+    {
+        car.update();
+        auto currentDistance   = car.getDistance();
+        auto travelledDistance = initialDistance > currentDistance
+                                     ? initialDistance - currentDistance
+                                     : currentDistance - initialDistance;
+        hasReachedTargetDistance = travelledDistance >= smartcarlib::utils::getAbsolute(centimeters);
+    }
+    car.setSpeed(0);
+}
+
+void ctrlHeading(char input)
+{
+    switch (input)
+    {
+    case 'a': // rotate counter-clockwise going forward
+        car.setSpeed(forwardSpeed);
+        car.setAngle(leftDegrees);
+        forward = false;
+        break;
+    case 'd': // turn clock-wise
+        car.setSpeed(forwardSpeed);
+        car.setAngle(rightDegrees);
+        forward = false;
+        break;
+    case 'w': // go ahead
+        car.setSpeed(forwardSpeed);
+        car.setAngle(0);
+        obstacleAvoidance(BrakeDistance);
+        forward = true;
+        break;
+    case 's': // go back
+        if(car.getSpeed() != 0) {
+            carBrake();
+        }
+        else {
+            car.setSpeed(backwordSpeed);
+            car.setAngle(0);
+        }   
+        forward = false;
+        break;
+    default: // if you receive something that you don't know, just stop
+        car.setSpeed(0);
+        car.setAngle(0);
+    }
+}
+
+
+void setup()
+{
+    Serial.begin(9600);
+}
+
+void loop()
+{
+    // car.disableCruiseControl();
+    if (Serial.available())
+    {
+        char input = Serial.read(); // read everything that has been received so far and log down
+                                    // the last entry
+        ctrlHeading(input);
+        if(input == 'c')
+        {
+            cruiseControl();
+        }
+    }
+
+    if(updateFlag) 
```

Comment:
What is the point of the update flag? Nothing bad happens if you call update and not in cruise control mode.